oauth2/oauth2-jose/src/main/java/org/springframework/security/oauth2/jwt/NimbusJwtDecoder.java

```diff
@@ -309,6 +328,33 @@ private static URL toURL(String url) {
 			}
 		}
 
+		private static class CacheJWKSetCache implements JWKSetCache {
+
+			private final Cache jwkSetCache;
+			private final String cacheKey;
+
+			CacheJWKSetCache(Cache jwkSetCache) {
+				Assert.notNull(jwkSetCache, "jwkSetCache cannot be null");
+				this.jwkSetCache = jwkSetCache;
+				this.cacheKey = UUID.randomUUID().toString();
```

Comment:
Using a GUID for the key may make it tricky to work with outside of this class. Since the idea here is that you can also use this cache in other places in the code, I think that the JWK Set URL makes more sense as the key.

This also aligns with the sensibility that when caching a method, it's usually keyed by its parameters.